main.go

```diff
@@ -59,13 +61,35 @@ var (
 
 	// flags for external node
 	providerURI = fs.String("provider", "", "an URI pointing at a provider")
+
+	useNimbus = fs.Bool("nimbus", false, "Use Nimbus node")
 )
 
+// XXX: Consider isolating this with compiler flag or runtime flag.
+// If that doesn't work, and it causes issues for some reason, we can isolate
+// this change into a patch and have a custom build script for Nimbus.
+// Arrange that main.main runs on main thread.
+func init() {
+	runtime.LockOSThread()
+}
+
 func main() {
 	if err := ff.Parse(fs, os.Args[1:]); err != nil {
 		exitErr(errors.Wrap(err, "failed to parse flags"))
 	}
 
+	// XXX: Spike, useNimbus stuff
+	if *useNimbus {
+		// do Nimbus stuff
+		nprocs := runtime.GOMAXPROCS(0)
```

Comment:
This has no effect.

Reply:
Right, this was for debugging thread stuff, can probably remove now

Reply:
Still useful for debugging though with new changes to Nimbus so I'd prefer to leave it in for now, thread gotchas